### ISSUE TITLE
OLLAMA_FLASH_ATTENTION regression on 0.3.10?

### ISSUE THREAD

**Issue by coodoo** [ollama/ollama#6769]
### What is the issue?

After upgrading to the latest version `0.3.10`, with `OLLAMA_FLASH_ATTENTION=1` set in env, seemed the tokens per second were halved, in my experiment, same code used to have tps around 23 and now it's only 11.

Wondering is there any known regression with regard to FLASH_ATTENTION?

### OS

macOS

### GPU

Apple

### CPU

Apple

### Ollama version

0.3.10

**Comment by jmorganca** [comment]
@coodoo is this on Apple Silicon? I'm still seeing a ~10% speedup between `OLLAMA_FLASH_ATTENTION=1` and `OLLAMA_FLASH_ATTENTION=0`.

Also, which model are you using?

Sorry you hit this, and thanks for creating an issue!

**Comment by coodoo** [comment]
Yup same M1 mac book pro, same model (`gemma2-9B`), same code, same test data.

I'm currently running more benchmarks to identify the root cause, hopefully something will come out! 🤞

**Comment by jmorganca** [comment]
This should be improved but let me know if it's not
